src/ident/Ident.h

```diff
@@ -9,17 +9,23 @@
 #ifndef SQUID_SRC_IDENT_IDENT_H
 #define SQUID_SRC_IDENT_IDENT_H
 
-#if USE_IDENT
-
 #include "cbdata.h"
 #include "comm/forward.h"
+#include "sbuf/forward.h"
 
-typedef void IDCB(const char *ident, void *data);
+#include <optional>
 
 /// Ident Lookup API
 namespace Ident
 {
 
+/// A user-id field of an RFC 931 auth-reply response.
```

Comment:
```suggestion
/// A user-id field of an RFC 1413 auth-reply response.
```

BTW, please note that this newer RFC (that we ought to be using because it obsoleted RFC 931) requires user-id to be non-empty:

```
<user-id> ::= <octet-string>
<octet-string> ::= 1*512<octet-characters>
```

Reply:
I replaced with RFC 1413 here and throughout the code (344847d).

> requires user-id to be non-empty

Do you suggest to fix Ident::ReadReply() accordingly? In this case, we would need something like state->deleteThis("missing user-id") there instead of state->notify() if user-id proves to be missing.

Reply:
> Do you suggest to fix Ident::ReadReply() accordingly?

```C++
if (strstr(buf, "USERID")) {
    if ((ident = strrchr(buf, ':'))) {
        while (xisspace(*++ident));
        if (ident && *ident == '\0')
            ident = nullptr;
        state->notify(ident);
    }
}
```

Is not the existing `\0` check (after at least one `++ident` increment) already enough to reject empty user-id values?

Reply:
I agree that we pass nullptr instead of '\0' to notify but there is no indication of a parsing error, that is why I suggested adding deleteThis() with a message. BTW, we can remove the 'ident' check in that if statement because it cannot drop to zero after incrementation.

Reply:
As discussed during a meeting, let's refactor to provide fault-specific deleteThis() calls for missing info.

Reply:
Done (ac20a49).